src/main/java/com/epam/test_generator/services/StepSuggestionService.java

```diff
@@ -144,4 +145,27 @@ public void removeStepSuggestion(Long stepSuggestionId) {
 
         stepSuggestionDAO.delete(stepSuggestionId);
     }
+
+    /**
+     * Find all steps suggestions by string
+     *
+     * @param searchString - string for search
+     * @return list with found steps suggestions
+     */
+    public List<StepSuggestionDTO> findStepsSuggestions(String searchString) {
+        if (searchString.length() < 4) {
+            throw new BadRequestException("Search string has to be longer than 3 characters!");
+        }
+        List<StepSuggestion> allStepsSuggestions = stepSuggestionDAO.findAll();
```

Comment:
https://www.postgresql.org/message-id/200305251258.54689.josh%40agliodbs.com

Your search query would be **SELECT id,content, type FROM STEPSUGGESTION where content ILIKE '%***value***%'**

ILIKE is for case-insensitive search

Reply:
Yes, I agree. This is a more correct approach. But probably will be better to write a custom spring jpa method, it will be more readable and won't depend on the database.